### PR TITLE
Use the builtin GTK3 FileChooser rather than our custom subclass.

### PR DESCRIPTION
**Pull request by anntzer**
## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

File: lib/matplotlib/backends/backend_gtk3.py [hunk 1]
```diff
@@ -529,18 +532,48 @@ def get_filechooser(self):
...
+        dialog.set_current_name(self.canvas.get_default_filename())
+        dialog.set_do_overwrite_confirmation(True)
+
+        response = dialog.run()
+        fname = dialog.get_filename()
+        ff = dialog.get_filter()  # Doesn't autoadjust to filename :/
+        fmt = self.canvas.get_supported_filetypes_grouped()[ff.get_name()][0]
+        dialog.destroy()
+        if response == Gtk.ResponseType.CANCEL:
+            return
```
Comment (QuLogic) [inline lib/matplotlib/backends/backend_gtk3.py]: Probably can do this earlier?

File: lib/matplotlib/backends/backend_gtk3.py [hunk 2]
```diff
@@ -529,18 +532,48 @@ def get_filechooser(self):
...
+                dialog.set_filter(ff)
+
+        @functools.partial(dialog.connect, "notify::filter")
+        def on_notify_filter(*args):
+            name = dialog.get_filter().get_name()
+            fmt = self.canvas.get_supported_filetypes_grouped()[name][0]
+            dialog.set_current_name(
+                str(Path(dialog.get_current_name()).with_suffix("." + fmt)))
+
+        dialog.set_current_folder(rcParams["savefig.directory"])
```
Comment (QuLogic) [inline lib/matplotlib/backends/backend_gtk3.py]: Current recommendations are to not even call this.

File: lib/matplotlib/backends/backend_gtk3.py [hunk 3]
```diff
@@ -529,18 +532,48 @@ def get_filechooser(self):
...
-        if fname:
-            startpath = os.path.expanduser(rcParams['savefig.directory'])
-            # Save dir for next time, unless empty str (i.e., use cwd).
-            if startpath != "":
-                rcParams['savefig.directory'] = os.path.dirname(fname)
-            try:
-                self.canvas.figure.savefig(fname, format=format)
-            except Exception as e:
-                error_msg_gtk(str(e), parent=self)
+        dialog = Gtk.FileChooserDialog(
```
[the remark this answers is not in the document]
Comment (anntzer) [reply lib/matplotlib/backends/backend_gtk3.py]: The problem with FileChooserNative (when it's using a Gtk FileChooser) is that the notify::filter callback is not called, and then it's easy to end up with a different "apparent" extension from the one set in the file format combobox.  Right now we use the combobox value, so if one types in foo.pdf but the combobox says "PNG", they get a png file called "foo.pdf"...
(Compare with KDialog, which keeps the two in sync by default, without needing the custom callback, and with a tickbox to disable that behavior.)

File: lib/matplotlib/backends/backend_gtk3.py (hunk 1 above)
Comment (anntzer) [reply lib/matplotlib/backends/backend_gtk3.py]: get_filename() and get_filter() return None after destroy() is called, so if we move this up we'd need to duplicate the call to destroy() into the if clause.

File: lib/matplotlib/backends/backend_gtk3.py (hunk 2 above)
Comment (anntzer) [reply lib/matplotlib/backends/backend_gtk3.py]: I know, but that'd mean not respecting the documented rcParam...

**Comment by anntzer** [comment]
should be fixed now

**Pull request merged**